File: src/main/java/com/afollestad/polarupgradetool/jfx/UICallback.java
package com.afollestad.polarupgradetool.jfx;

/**
 * Project : polar-dashboard-upgrade-tool
 * Author : pddstudio
 * Year : 2016
 */
public interface UICallback {

    void onProjectDetected(String applicationName, String applicationPackage, String applicationVersionName, String applicationVersionCode);

    void onErrorOccurred(String errorMessage);

    void onArchiveDownloadStarted(String sizeStr);

    void onArchiveDownloadProgress(String progressStr);

    void onArchiveDownloadFailed(String errorMessage);

    void onArchiveDownloadSuccess();

    void onStatusUpdate(String statusMessage);

    void onUpdateSuccessful();
}

File: src/main/java/com/afollestad/polarupgradetool/jfx/UICallback.kt
package com.afollestad.polarupgradetool.jfx

/**
 * Project : polar-dashboard-upgrade-tool
 * Author : pddstudio
 * Year : 2016
 */
interface UICallback {

    fun onProjectDetected(applicationName: String, applicationPackage: String, applicationVersionName: String, applicationVersionCode: String)

    fun onErrorOccurred(errorMessage: String)

    fun onArchiveDownloadStarted(sizeStr: String)

    fun onArchiveDownloadProgress(progressStr: String)

    fun onArchiveDownloadFailed(errorMessage: String)

    fun onArchiveDownloadSuccess()

    fun onStatusUpdate(statusMessage: String)

    fun onUpdateSuccessful()
}